### PR TITLE
Restructure the top-level directory

### PR DESCRIPTION
**Pull request by chrislessard**
Hi,

This series of commit restructures the project hierarchy as discussed in our meeting on Wednesday, January 23rd 2019. 

Solves #51

### CODE REVIEW

**Review by dingtaoliu — COMMENTED** [review]
Most of the changes are just moving files around, the few code changes (such as changing proxy settings on the frontend) all make sense to me.  My only concern (I'm not too familiar with git) is whether or not to wait until the current branches are merged before merging this?  I'm not sure if this change will cause a bunch of conflicts when merging the current branches because the files that were changed will have been moved to a new location.

**Review by Yifang-Zhang — APPROVED** [review]

**Comment by chrislessard** [comment]
@dingtaoliu Fair enough but it really doesnt matter too much. Personally I would prefer if you guys rebase, since you know which files you changed and because it would mean I wouldn't have to rebase all of it, but its no big deal if I have to !

**Comment by chrislessard** [comment]
Closing, please see https://github.com/uoft-tapp/tapp/compare/TAPP-51-b?expand=1

**Pull request closed without merging**
